### PR TITLE
Allow playing custom streams

### PR DESCRIPTION
**Pull request by jacksonh**
Opening this up just to see if you are OK with the approach, I still have some refactoring to do and will add an example/test. 

This lets users implement custom streams that can be played with AudioStreaming. For example, I have a websocket interface that I fetch data from. I can wrap that stream into a CoreAudioStreamSource and add that to the player. Another example would be decrypting data.

### CODE REVIEW

**Comment by dimitris-c** [comment]
Nice one, thanks. Could you write a simple custom CoreAudioStreamSource on the example project? Just to verify the behaviour. And I guess the same API can be used for the `queue` method.

**Review by dimitris-c — APPROVED** [review]

**Comment by jacksonh** [comment]
FYI, just fleshing out my use of this more in our app to see if there's anything I'd want to change. Also still plan on completing the queue methods.

**Comment by jacksonh** [comment]
Took a bit to get to that, but using in my app now with this branch and all seems to be working well.

File: AudioPlayer/AudioPlayer/Content/AudioPlayer/AudioPlayerControlsView.swift [hunk 1]
```diff
@@ -247,11 +248,23 @@ extension AudioPlayerControls {
...
+                    audioPlayerService.play(source: source, entryId: track.url.absoluteString, format: audioFormat)
+                    currentTrack = track
+                } else {
+                    audioPlayerService.play(url: track.url)
+                }
             }
         }
 
+        func createStreamSource() -> CoreAudioStreamSource {
+            return CustomStreamAudioSource(underlyingQueue: audioPlayerService.player.sourceQueue)
```
Comment (dimitris-c) [inline AudioPlayer/AudioPlayer/Content/AudioPlayer/AudioPlayerControlsView.swift:265]: I don't like this bit, passing the inner sourceQueue from the AudioPlayer itself, but I know why it's required.
Can't think of a quick way of getting this from the AudioPlayer.

**Comment by jacksonh** [comment]
let me know if you want me to add anything else here. Trying to figure out a way to test better.

**Comment by dimitris-c** [comment]
Released on 1.2.6

**Pull request merged**